semantic-analysis/src/Analysis/Carrier/Statement/State.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+module Analysis.Carrier.Statement.State
+( -- * Messages
+  Message(..)
+  -- * Statement carrier
+, runStatement
+, StatementC(..)
+  -- * Statement effect
+, module Analysis.Effect.Statement
+) where
+
+import           Analysis.Effect.Statement hiding (Import)
+import qualified Analysis.Effect.Statement as S
+import           Control.Algebra
+import           Control.Carrier.State.Church
+import           Control.Monad.Fail as Fail
+import           Data.List.NonEmpty (NonEmpty)
+import           Data.Text (Text)
+
+-- Messages
+
+data Message
```

Comment:
newtype here?

Reply:
Expectation is that there'll be other message types, but in  the meantime, yeah, good call.